src/textual/widgets/_input.py

```diff
@@ -226,9 +241,13 @@ def __init__(
         highlighter: Highlighter | None = None,
         password: bool = False,
         *,
+        restrict: str | None = None,
+        type: str = "text",
```

Comment:
If this takes one of a specific set of types, perhaps make this a `Literal` so that it allows for type checking and completion?